### DISCUSSION TITLE
increment_version_number by app store lookup

### DISCUSSION THREAD

**Discussion by reshadf** [fastlane/fastlane#17417] [Ideas]
so we have been trying to implement increment_version_number and increment_build_number via Fastlane. However, there are so many implementations that there is no clear way of "automating" This results in a big team of devs to always commit the "marketing version" manually before starting a new release or use some other script/tool to do this. 

I was thinking, why do we not get the current released version from the app store in order to update this?

**Comment by rogerluan** [comment]
Hey @reshadf 👋 

In the company I work at, we've concluded that it's much easier and safer to explicitly pass the version being released as an argument when deploying. The reason is that you are the person who knows the best which version should be the next version to be released, i.e. whether it's gonna be a patch release, a minor, or a major release.
There're ways to identify that via e.g. PR labeling and such, but it's just added complexity that we found unnecessary and would be considered over engineering.

Our build number is automatically during build phase (it's an auto generated number that has guaranteed uniqueness), so we don't have to manually bump it upon releasing.

>why do we not get the current released version from the app store in order to update this?

This can be achieved by using _spaceship_'s available functions, you can call them directly from ruby/Fastfile, if you think it better fits your needs 😊 

Hope this other point of view help you out 🙌
